azure_sdk/lib/azure_sdk/client.rb

```diff
@@ -0,0 +1,165 @@
+        :datalake_store_account => 'Azure::ARM::DataLakeStore::Account::DataLakeStoreAccountManagementClient',
+        :devtestlabs => 'Azure::ARM::DevTestLabs::DevTestLabsClient',
+        :dns => 'Azure::ARM::Dns::DnsManagementClient',
+        :features => 'Azure::ARM::Features::FeatureClient',
+        :graph_rbac => 'Azure::ARM::Graph::GraphRbacManagementClient',
+        :iot_hub => 'Azure::ARM::IotHub::IotHubClient',
+        :key_vault => 'Azure::ARM::KeyVault::KeyVaultManagementClient',
+        :locks => 'Azure::ARM::Locks::ManagementLockClient',
+        :logic => 'Azure::ARM::Logic::LogicManagementClient',
+        :machine_learning => 'Azure::ARM::MachineLearning::AzureMLWebServicesManagementClient',
+        :media_services => 'Azure::ARM::MediaServices::MediaServicesManagementClient',
+        :mobile_engagement => 'Azure::ARM::MobileEngagement::EngagementManagementClient',
+        :network => 'Azure::ARM::Network::NetworkManagementClient',
+        :notification_hubs => 'Azure::ARM::NotificationHubs::NotificationHubsManagementClient',
+        :policy => 'Azure::ARM::Policy::PolicyClient',
+        :powerbi_embedded => 'Azure::ARM::PowerBiEmbedded::PowerBIEmbeddedManagementClient',
+        :redis => 'Azure::ARM::Redis::RedisManagementClient',
+        :resources => 'Azure::ARM::Resources::ResourceManagementClient',
+        :scheduler => 'Azure::ARM::Scheduler::SchedulerManagementClient',
+        :search => 'Azure::ARM::Search::SearchManagementClient',
+        :server_management => 'Azure::ARM::ServerManagement::ServerManagement',
+        :service_bus => 'Azure::ARM::ServiceBus::ServiceBusManagementClient',
+        :sql => 'Azure::ARM::SQL::SqlManagementClient',
+        :storage => 'Azure::ARM::Storage::StorageManagementClient',
+        :subscription => 'Azure::ARM::Subscriptions::SubscriptionClient',
+        :traffic_manager => 'Azure::ARM::TrafficManager::TrafficManagerManagementClient',
+        :web => 'Azure::ARM::Web::WebSiteManagementClient'
+    }
+
+    #
+    # Initializes a new instance of the Client class.
+    # @param options [Hash] hash of client options.
+    #
+    def initialize(options = {})
+      if options.is_a?(Hash) && options.length == 0
+        @options = setup_options
+      else
+        @options = options
+      end
+
+      reset!(options)
+
+      #
+      # Dynamically create arm client methods
+      #
+      @@supported_clients.each do |key, value|
+        create_arm_client_methods(key, value)
+      end
+    end
+
+    #
+    # Check if this client is configured with the same options.
+    # @param opts [Hash] hash of client options.
+    # @return [Boolean] True if client is configured with same set of options, False otherwise.
+    #
+    def same_options?(opts)
+      opts.hash == @options.hash
+    end
+
+    #
+    # Check if this client is configured with the some options.
```

Comment:
Yup. Corrected.

---
In reply to: [76343363](https://github.com/Azure/azure-sdk-for-ruby/pull/433#discussion_r76343363) [](ancestors = 76343363)